guppylang/cfg/bb.py

```diff
@@ -66,10 +60,10 @@ class BB(ABC):
     branch_pred: ast.expr | None = None
 
     # Information about assigned/used variables in the BB
-    _vars: VariableStats | None = None
+    _vars: VariableStats[str] | None = None
 
     @property
-    def vars(self) -> VariableStats:
+    def vars(self) -> VariableStats[str]:
```

Comment:
`bb.vars` is still used to look up usage locations for error locations, e.g. in `checker/cfg_checker.py`.

But agreed, we can get rid of this now. However, I'd prefer doing this refeactor in a separate PR, otherwise we'll get annoying merge conflicts with the other PRs stacked on top of this one :D

I created issue #305 for this